src/dos/cdrom_image.cpp

```diff
@@ -154,8 +248,10 @@ CDROM_Interface_Image::~CDROM_Interface_Image()
 	if (player.cd == this) player.cd = NULL;
 	ClearTracks();
 	if (refCount == 0) {
-		SDL_DestroyMutex(player.mutex);
-		player.channel->Enable(false);
+		StopAudio();
+		MIXER_DelChannel(player.channel);
+		player.channel = NULL;
```

Comment:
`nullptr` ;) ?

Reply:
Fixed in 17c97348, also updated mp3.cpp; the two C++ files part of the patch.

Reply:
This fix rippled a bit :-)

Switching NULL to nullptr produced: _"cdrom_image.cpp:70:14: error: ‘nullptr’ was not declared in this scope"_

Turns out nullptr is only recognized in C++03 and above. Fortunately there's an existing way to  check for it in autotools, which I've added here: https://github.com/dreamer/dosbox-staging/commit/6468970939c4b95bb5eab4ccac006b7aeca8ab24

Tested the change locally and worked great.  However, in CI `./configure` fails.  Turns out that configure.ac statement `AX_CXX_COMPILE_STDCXX_0X` is actually an m4 macro-file that doesn't exist by default.

http://git.savannah.gnu.org/gitweb/?p=autoconf-archive.git;a=blob_plain;f=m4/ax_cxx_compile_stdcxx_0x.m4

After a bit more digging, I see that the package `autoconf-archive` supplies it. It was also a relief to see that this package is available across the board in all our CI package managers.